Sources/BrowserServicesKit/RemoteMessaging/Mappers/JsonRemoteMessageMapper.swift

```diff
@@ -0,0 +1,211 @@
+            guard let primaryActionText = content.primaryActionText,
+                  !primaryActionText.isEmpty,
+                  let action = mapToAction(content.primaryAction)
+            else {
+                return nil
+            }
+
+            return .bigSingleAction(titleText: content.titleText,
+                                    descriptionText: content.descriptionText,
+                                    placeholder: mapToPlaceholder(content.placeholder),
+                                    primaryActionText: primaryActionText,
+                                    primaryAction: action)
+        case .bigTwoAction:
+            guard let primaryActionText = content.primaryActionText,
+                  !primaryActionText.isEmpty,
+                  let primaryAction = mapToAction(content.primaryAction),
+                  let secondaryActionText = content.secondaryActionText,
+                  !secondaryActionText.isEmpty,
+                  let secondaryAction = mapToAction(content.secondaryAction)
+            else {
+                return nil
+            }
+
+            return .bigTwoAction(titleText: content.titleText,
+                                 descriptionText: content.descriptionText,
+                                 placeholder: mapToPlaceholder(content.placeholder),
+                                 primaryActionText: primaryActionText,
+                                 primaryAction: primaryAction,
+                                 secondaryActionText: secondaryActionText,
+                                 secondaryAction: secondaryAction)
+        case .none:
+            return nil
+        }
+    }
+    // swiftlint:enable function_body_length
+
+    static func mapToAction(_ jsonAction: RemoteMessageResponse.JsonMessageAction?) -> RemoteAction? {
+        guard let jsonAction = jsonAction else {
+            return nil
+        }
+
+        switch RemoteMessageResponse.JsonActionType(rawValue: jsonAction.type) {
+        case .url:
+            return .url(value: jsonAction.value)
+        case .appStore:
+            return .appStore
+        case .dismiss:
+            return .dismiss
+        case .none:
+            return nil
+        }
+    }
+
+    static func mapToPlaceholder(_ jsonPlaceholder: String?) -> RemotePlaceholder {
+        guard let jsonPlaceholder = jsonPlaceholder else {
+            return .announce
+        }
+
+        switch RemoteMessageResponse.JsonPlaceholder(rawValue: jsonPlaceholder) {
+        case .announce:
+            return .announce
+        case .appUpdate:
+            return .appUpdate
+        case .ddgAnnounce:
+            return .ddgAnnounce
+        case .criticalUpdate:
+            return .criticalUpdate
+        case .none:
+            return .announce
+        }
+    }
+
+    // swiftlint:disable cyclomatic_complexity
+    static func maps(jsonRemoteRules: [RemoteMessageResponse.JsonMatchingRule]) -> [Int: [MatchingAttribute]] {
```

Comment:
This function feels like a lot of boilerplate to me that I'n not sure is necessary, especially as for each case it does a very similar call out to JsonRulesMapper, which then seems to do the same boiler plate in each case